### PR TITLE
Add HLRC docs for Get Lifecycle Policy

### PR DESCRIPTION
**Pull request by AthenaEryma**
Adds docs for the Get Lifecycle Policy API to the HLRC documentation.

### CODE REVIEW

File: client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/ILMDocumentationIT.java [hunk 1]
```diff
@@ -119,6 +123,92 @@ public void onFailure(Exception e) {
...
+        ImmutableOpenMap<String, LifecyclePolicyMetadata> policies =
+            response.getPolicies();
+        LifecyclePolicyMetadata myPolicyMetadata =
+            policies.get("my_policy"); // <1>
+        String myPolicyName = myPolicyMetadata.getName();
+        long version = myPolicyMetadata.getVersion();
+        String lastModified = myPolicyMetadata.getModifiedDateString();
+        long lastModifiedDate = myPolicyMetadata.getModifiedDate();
+        LifecyclePolicy myPolicy = myPolicyMetadata.getPolicy(); // <2>
+        // end::ilm-get-lifecycle-policy-response
```
Comment (colings86) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/ILMDocumentationIT.java]: Can we add some asserts to ensure we have retrieved the correct thing from the API. Like checking that the policy we got back matches the policy we sent?

File: client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/ILMDocumentationIT.java [hunk 2]
```diff
@@ -119,6 +123,92 @@ public void onFailure(Exception e) {
...
+            phases.put("delete",
+                new Phase("delete",
+                    new TimeValue(90, TimeUnit.DAYS), deleteActions));
+
+            LifecyclePolicy policy = new LifecyclePolicy("my_policy",
+                phases);
+            PutLifecyclePolicyRequest putRequest =
+                new PutLifecyclePolicyRequest(policy);
+            LifecyclePolicy policy2 = new LifecyclePolicy("other_policy",
+                phases);
```
Comment (colings86) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/ILMDocumentationIT.java]: Could we make the two policies slightly different (even if its only the min age on the phase so we can differentiate them when we assert the policy that is returned later?

**Review by colings86 — COMMENTED** [review]
I left some minor comments about adding asserts. I think this is important because without the asserts we aren't actually checking the code we are documenting is doing what we expect.

**Review by talevy — APPROVED** [review]
LGTM

**Pull request merged**